### PR TITLE
Add docstrings to classes

### LINKED ISSUE #18: Add doc strings to method annotation classes
Method annotation classes in `uplink/decorators.py` are missing class doc strings. To improve code documentation, we need to add doc strings to the following classes, adhering the [Google Style Guide](https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments) for consistency with the rest of the codebase:

- [ ] `uplink.decorators.headers`
- [ ] `uplink.decorators.form_url_encode`
- [ ] `uplink.decorators.multipart`
- [ ] `uplink.decorators.json`
- [ ] `uplink.decorators.timeout`
- [ ] `uplink.decorators.args`

### PR DESCRIPTION
**Pull request by FabioRosado**
Fixes #18 .

Changes proposed in this pull request:
- Adds docstrings to method classes: 
  - headers
  -  json
  - timeout
  - form_url_encode
  - multipart
  - args

I've attempted to write some docstrings following the Google Python Style Guide. I've used uplink documentation to write the docstrings, but there are some bits that I'm uncertain of so I'm making this PR just to get some feedback on this.

Should I add a docstring to the return class as well just to explain what it does?
Thank you for the time and as I said would appreciate any feedback! 

Attention: @prkumar

### CODE REVIEW

File: uplink/decorators.py [hunk 1]
```diff
@@ -106,11 +108,23 @@ def __init__(self, arg, **kwargs):
...
 # noinspection PyPep8Naming
 class form_url_encoded(MethodAnnotation):
+    """URL-encodes the request body.
+
+    Used on POST/PUT/PATCH request. It url-encodes the body
+    of the message and sets the header to:
+    `Content-Type:application/x-www-form-urlencoded`
+
+    Attributes:
+        can_be_static = A boolean value indicating that this
```
Comment (prkumar) [inline uplink/decorators.py]: Instead of equals (`can_be_static =`) use colon (`can_be_static :`).

Comment (prkumar) [inline uplink/decorators.py]: Instead of equals (`can_be_static =`) use colon (`can_be_static :`).

File: uplink/decorators.py [hunk 2]
```diff
@@ -106,11 +108,23 @@ def __init__(self, arg, **kwargs):
...
+        annotation can be used as a decorator without any
```
Comment (prkumar) [inline uplink/decorators.py]: Nit: indent continuing lines.

File: uplink/decorators.py [hunk 3]
```diff
@@ -94,9 +94,11 @@ def modify_request(self, request_builder):
 
 # noinspection PyPep8Naming
 class headers(MethodAnnotation):
-    """
-    You can apply this decorator on a class to add headers to all of
-    its API definitions.
+    """Use as a decorator on a class to add Headers to all API calls.
+
+    Attributes:
+        _headers:  A dictionary containing header field as Key
```
Comment (prkumar) [inline uplink/decorators.py]: In the class docstrings, let's omit the attributes if they start with leading underscores, which indicates they should be treated as private methods.

File: uplink/decorators.py [hunk 4]
```diff
@@ -94,9 +94,11 @@ def modify_request(self, request_builder):
...
+                   and header values as Value
     """
```
Comment (prkumar) [inline uplink/decorators.py]: Further, when the class has a constructor, let's document the constructor arguments in the class docstring. Checkout [this guide](http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html), which shows different examples using the Google Python Style Guide. Specifically, take a look at the `ExampleClass`.

Comment (FabioRosado) [reply uplink/decorators.py]: I've replaced with something like this:
```
    """Use as a decorator on a class to add Headers to all API calls.

    Args:
        arg: A dict containing header input parameters
        kwargs: A dict containing header input parameters
    """
```
Is this what you meant? Thanks for the feedback!

Comment (prkumar) [reply uplink/decorators.py]: Yes! Looks good!

**Comment by FabioRosado** [comment]
I've updated the docstrings following your suggestions. What do you think about it now?

**Comment by prkumar** [comment]
@FabioRosado Also, may I add the name and username listed on your GitHub profile to `AUTHORS.rst`? If you want your entry to display a different name or social media handle, let me know, or you can open a PR with the addition!

**Comment by FabioRosado** [comment]
@prkumar  I forgot to do that, apologies! I've commented on issue 17 so I'll add myself to the authors when I finish working on that issue 👍

**Pull request merged**